controllers/register.js

```diff
@@ -0,0 +1,75 @@
+const uuid = require('uuid/v4');
+const passport = require('passport');
+const mongoose = require('mongoose');
+
+const {logger} = require('../logging/logger');
+const User = mongoose.model('User');
+const FacebookTokenStrategy = require('passport-facebook-token');
+const GoogleTokenStrategy = require('passport-google-token').Strategy;
+
+const FACEBOOK_APP_ID = '1650628351692070';
+const GOOGLE_APP_ID = '12899066904-jqhmi5uhav530aerctj631gltumqvk8i.apps.googleusercontent.com';
+const AUTH_TYPES = {
+    FACEBOOK: 'facebook',
+    GOOGLE: 'google',
+};
+
+passport.use(new FacebookTokenStrategy({
```

Comment:
Does passport validate that the user is a valid Facebook user, and that the user is a member of our Facebook app? For the time being, we only want known users who are registered with our organization to be able to auth to our API

Reply:
Afaik the access token does not contain data about the application. Basically the flow goes the user logs into the app on their mobile device, gets a oauth access token, which it then passes to the register endpoint. The register endpoint validates the access token, and creates a user id and api key that the logged in user can then use to make api requests